src/raftstore/store/worker/apply.rs

```diff
@@ -0,0 +1,321 @@
+                             notify_stale_command};
+
+pub struct Apply {
+    region_id: u64,
+    term: u64,
+    entries: Vec<Entry>,
+}
+
+pub struct Read {
+    region_id: u64,
+    term: u64,
+    cmd: RaftCmdRequest,
+    cb: Callback,
+}
+
+pub struct Registration {
+    id: u64,
+    term: u64,
+    apply_state: RaftApplyState,
+    applied_index_term: u64,
+    region: Region,
+}
+
+impl Registration {
+    fn new(peer: &Peer) -> Registration {
+        Registration {
+            id: peer.peer_id(),
+            term: peer.term(),
+            apply_state: peer.get_store().apply_state.clone(),
+            applied_index_term: peer.get_store().applied_index_term,
+            region: peer.region().clone(),
+        }
+    }
+}
+
+pub struct Propose {
+    id: u64,
+    region_id: u64,
+    is_conf_change: bool,
+    uuid: Uuid,
+    term: u64,
+    cb: Callback,
+}
+
+pub struct Destroy {
+    region_id: u64,
+}
+
+/// region related task.
+pub enum Task {
+    Apply(Apply),
+    Read(Read),
+    Registration(Registration),
+    Propose(Propose),
+    Destroy(Destroy),
+}
+
+impl Task {
+    pub fn apply(region_id: u64, term: u64, entries: Vec<Entry>) -> Task {
+        Task::Apply(Apply {
+            region_id: region_id,
+            term: term,
+            entries: entries,
+        })
+    }
+
+    pub fn read(region_id: u64, term: u64, cmd: RaftCmdRequest, cb: Callback) -> Task {
+        Task::Read(Read {
+            region_id: region_id,
+            term: term,
+            cmd: cmd,
+            cb: cb,
+        })
+    }
+
+    pub fn register(peer: &Peer) -> Task {
+        Task::Registration(Registration::new(peer))
+    }
+
+    pub fn propose(id: u64,
+                   region_id: u64,
+                   uuid: Uuid,
+                   is_conf_change: bool,
+                   term: u64,
+                   cb: Callback)
+                   -> Task {
+        Task::Propose(Propose {
+            id: id,
+            region_id: region_id,
+            uuid: uuid,
+            term: term,
+            is_conf_change: is_conf_change,
+            cb: cb,
+        })
+    }
+
+    pub fn destroy(region_id: u64) -> Task {
+        Task::Destroy(Destroy { region_id: region_id })
+    }
+}
+
+impl Display for Task {
+    fn fmt(&self, f: &mut Formatter) -> fmt::Result {
+        match *self {
+            Task::Apply(ref a) => write!(f, "[region {}] async apply", a.region_id),
+            Task::Propose(ref p) => write!(f, "[region {}] propose", p.region_id),
+            Task::Read(ref r) => write!(f, "[region {}] read", r.region_id),
+            Task::Registration(ref r) => {
+                write!(f, "[region {}] Reg {:?}", r.region.get_id(), r.apply_state)
+            }
+            Task::Destroy(ref d) => write!(f, "[region {}] destroy", d.region_id),
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct ApplyRes {
+    pub region_id: u64,
+    pub apply_state: RaftApplyState,
+    pub size_diff_hint: i64,
```

Comment:
Could you extract the four fields to an structure ? It would make line 222 looks better. Or you can mark the changes as TODO. size_diff_hint, delete_keys_hint, written_keys, written_bytes